src/rust/id.shinigami/src/lib.rs

```diff
@@ -0,0 +1,287 @@
+const CDN_URL: &str = "https://storage.shngm.id";
+const BASE_URL: &str = "https://app.shinigami.asia";
+
+// Todo: Search, Tags, Filters
+
+#[get_manga_list]
+fn get_manga_list(filters: Vec<Filter>, page: i32) -> Result<MangaPageResult> {
+    let mut url = format!(
+        "{}/v1/manga/list?type=project&page={}&page_size=30&is_update=true&sort=latest&sort_order=desc",
+        API_URL, page
+    );
+    
+    // Handle search and filters here
+    for filter in filters {
+        match filter.kind {
+            FilterType::Title => {
+                if let Ok(query) = filter.value.as_string() {
+                    url.push_str(&format!("&q={}", query.read()));
+                }
+            },
+            _ => continue,
+        }
+    }
+
+    let json = Request::new(url, HttpMethod::Get)
+        .header("Accept", "application/json")
+        .header("Origin", BASE_URL)
+        .header("DNT", "1")
+        .header("Sec-GPC", "1")
+        .json()?;
+
+    let data = json.as_object()?;
+    
+    if data.get("retcode").as_int()? != 0 {
+        return Ok(MangaPageResult { manga: Vec::new(), has_more: false });
+    }
+
+    let manga_array = data.get("data").as_array()?;
+    let mut mangas = Vec::new();
+
+    for manga in manga_array {
+        let manga_obj = manga.as_object()?;
+        let title = manga_obj.get("title").as_string()?.read();
+        let _alt_title = manga_obj.get("alternative_title").as_string()?.read();
+        
+        let cover = if let Ok(url) = manga_obj.get("cover_image_url").as_string() {
+            url.read()
+        } else {
+            manga_obj.get("cover_image_url").as_string()?.read()
+        };
+
+        let taxonomy = manga_obj.get("taxonomy").as_object()?;
+        let mut genres = Vec::new();
+        if let Ok(genre_array) = taxonomy.get("Genre").as_array() {
+            for genre in genre_array {
+                if let Ok(genre_obj) = genre.as_object() {
+                    if let Ok(genre_name) = genre_obj.get("name").as_string() {
+                        genres.push(genre_name.read());
+                    }
+                }
+            }
+        }
+
+        // Fix in get_manga_list function
+        mangas.push(Manga {
+            id: manga_obj.get("manga_id").as_string()?.read(),
+            title,
+            author: get_taxonomy_names(&taxonomy, "Author"),
+            artist: get_taxonomy_names(&taxonomy, "Artist"),
+            description: manga_obj.get("description").as_string()?.read(),
+            url: manga_obj.get("manga_id").as_string()?.read(),
+            cover,
+            categories: genres,
+            status: match manga_obj.get("status").as_int()? {
+                1 => MangaStatus::Ongoing,
+                2 => MangaStatus::Completed,
+                _ => MangaStatus::Unknown,
+            },
+            nsfw: MangaContentRating::Safe,
+            viewer: MangaViewer::Rtl
+            // Remove ..Default::default() here
+        });
```

Comment:
you shouldn't be fetching all this data in `get_manga_list`, just do it in `get_manga_details`. this function should only return manga with the title and cover populated. the rest can be filled in with `get_manga_details`.